app/connector/SchemeConnector.scala

```diff
@@ -59,9 +61,13 @@ class SchemeConnector @Inject()(
     ).map { response =>
       response.status match {
         case OK =>
+          val totalResults = (response.json \ "totalResults").toOption.flatMap(_.asOpt[Int]).getOrElse(0)
```

Comment:
This can be simplified as - (response.json \ "totalResults").asOpt[Int].getOrElse(0)

toOption and flatMap are essentially cancelling each other

Reply:
Well spotted! Fixed now